sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/OracleConnectionProvider.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.jdbc.connection
+
+import java.security.PrivilegedExceptionAction
+import java.sql.{Connection, Driver}
+import java.util.Properties
+
+import org.apache.hadoop.security.UserGroupInformation
+
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions
+
+private[sql] class OracleConnectionProvider(driver: Driver, options: JDBCOptions)
+    extends SecureConnectionProvider(driver, options) {
+  override val appEntry: String = "kprb5module"
+
+  override def getConnection(): Connection = {
+    setAuthenticationConfigIfNeeded()
+    UserGroupInformation.loginUserFromKeytabAndReturnUGI(options.principal, options.keytab).doAs(
+      new PrivilegedExceptionAction[Connection]() {
+        override def run(): Connection = {
+          OracleConnectionProvider.super.getConnection()
+        }
+      }
+    )
+  }
+
+  override def getAdditionalProperties(): Properties = {
+    val result = new Properties()
+    // This prop needed to turn on kerberos authentication in the JDBC driver. Please see:
+    // scalastyle:off line.size.limit
+    // https://docs.oracle.com/en/database/oracle/oracle-database/19/jajdb/oracle/jdbc/OracleConnection.html#CONNECTION_PROPERTY_THIN_NET_AUTHENTICATION_SERVICES
+    // scalastyle:on line.size.limit
+    result.put("oracle.net.authentication_services", "(KERBEROS5)");
```

Comment:
Just a question. Why do we use `(KERBEROS5)` instead of `(KERBEROS)` in the given link? The new value looks correct to me, but it's weird to give a link with a different content. It would be great if we can give a more accurate link about `(KERBEROS5)` (if possible).

Reply:
At the first glance that smells like a major issue. What I can think of is that somehow the value and the define (`AnoServices.AUTHENTICATION_KERBEROS5`) mixed up. I'll double check in the driver implementation w/ JD-GUI + re-testing it on cluster to avoid questions. It will take a while so asking some patience.

When I've designed this I've had the feeling these configs must be overridable for flexibility. Now it's clear how it could pay off.

Reply:
Hmmm, during the investigation I've found the following code in the JDBC driver implementation w/ JD-GUI:
```
public interface AnoServices {
...
  public static final String AUTHENTICATION_KERBEROS5 = "KERBEROS5";
...
```
Let's see how the test behaves.

Reply:
Now I'm pretty sure this is a documentation issue because of the following:
* The driver [documentation](https://docs.oracle.com/en/database/oracle/oracle-database/19/jjdbc/client-side-security.html#GUID-991705F7-C2C5-4BA9-85D1-32749AE2FF64) gives the following as an example:
```
...
    Connection conn = 
      (Connection)Subject.doAs(specificSubject, new PrivilegedExceptionAction()
        {
          public Object run()
          {
            Connection con = null;
            Properties prop = new Properties();
            prop.setProperty(AnoServices.AUTHENTICATION_PROPERTY_SERVICES, 
                             "(" + AnoServices.AUTHENTICATION_KERBEROS5 + ")");
...
```
and that said I've found `AUTHENTICATION_KERBEROS5 = "KERBEROS5";` in `AnoServices`.
* Cluster test has passed w/ `KERBEROS5` 

Checking the documentation whether it fits to this version...

Reply:
I've double checked and the version matches which makes the situation weird (there is an interface value in the JDBC driver impl which is not documented).
I think what we can state here is that `AnoServices` constant works fine and that must be provided to make it work (that said such pre-configured configs can be overriden).
I would remove the link and add this information if you agree. WDYT @dongjoon-hyun @maropu ?

Reply:
+1 for that. Thank you!

Reply:
Thanks for the investigation! +1

Reply:
Thanks guys! Processing the remaining suggestions.

Reply:
Fixed the comment.